pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -367,12 +371,19 @@ func (c *EtcdCertSignerController) populateSecret(secretName, secretNamespace st
 		}
 		return err
 	}
-	secret.Data = map[string][]byte{
-		"tls.crt": cert.Bytes(),
-		"tls.key": key.Bytes(),
+	if err := ensureTLSData(secret); err != nil {
+		secretCopy := secret.DeepCopy()
+		secretCopy.Data = map[string][]byte{
+			"tls.crt": cert.Bytes(),
+			"tls.key": key.Bytes(),
+		}
+		klog.Warningf("secret %s/%s does not have valid data", secretNamespace, secretName)
+		klog.Infof("attempting to update secret %s/%s with valid certs", secretNamespace, secretName)
+		_, err = c.clientset.CoreV1().Secrets(secretNamespace).Update(secretCopy)
+		return err
 	}
-	_, err = c.clientset.CoreV1().Secrets(secretNamespace).Update(secret)
-	return err
+	klog.Infof("secret %s/%s has valid certs", secretNamespace, secretName)
```

Comment:
how chatty is this logging?

Reply:
This will only be enabled if the operator is started with INFO, honestly I am not sure about it.

IMO the convention is to log the errors with Errorf, and other things with a combination of Debugf, Infof and Warningf. This felt like an Infof to me.

I will read up on the log levels of klog and get back with a more precise answer

Reply:
I think we can adjust if it's too chatty.

Reply:
sure, marking it resolved for now